new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSessionMigration.java

```diff
@@ -0,0 +1,446 @@
+  private static final String SESSMIGR_IMAGE_NAME = "mii-image";
+
+  // constants for web service
+  private static final String SESSMIGR_APP_NAME = "sessmigr-app";
+  private static final String SESSMIGR_APP_WAR_NAME = "sessmigr-war";
+  private static final int SESSION_STATE = 4;
+  private static Map<String, String> httpAttrMap;
+
+  // constants for Operator and WebLogic domain
+  private static String domainUid = "sessmigr-domain-1";
+  private static String clusterName = "cluster-1";
+  private static String adminServerPodName = domainUid + "-admin-server";
+  private static String managedServerPrefix = domainUid + "-managed-server";
+  private static int managedServerPort = 8001;
+  private static int replicaCount = 2;
+  private static String opNamespace = null;
+  private static String domainNamespace = null;
+  private static ConditionFactory withStandardRetryPolicy = null;
+
+  /**
+   * Install Operator, create a custom image using model in image with model files
+   * and create a one cluster domain.
+   *
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   *                   JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void init(@Namespaces(2) List<String> namespaces) {
+    // create standard, reusable retry/backoff policy
+    withStandardRetryPolicy = with().pollDelay(2, SECONDS)
+      .and().with().pollInterval(10, SECONDS)
+      .atMost(5, MINUTES).await();
+
+    // get a unique Operator namespace
+    logger.info("Get a unique namespace for Operator");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    opNamespace = namespaces.get(0);
+
+    // get a unique domain namespace
+    logger.info("Get a unique namespace for WebLogic domain");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    domainNamespace = namespaces.get(1);
+
+    // install and verify Operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // create and verify WebLogic domain image using model in image with model files
+    String imageName = createAndVerifyDomainImage();
+
+    // create and verify one cluster domain
+    logger.info("Create domain and verify that it's running");
+    createAndVerifyDomain(imageName);
+
+    // map to save HTTP response data
+    httpAttrMap = new HashMap<String, String>();
+    httpAttrMap.put("sessioncreatetime", "(.*)sessioncreatetime>(.*)</sessioncreatetime(.*)");
+    httpAttrMap.put("sessionid", "(.*)sessionid>(.*)</sessionid(.*)");
+    httpAttrMap.put("primary", "(.*)primary>(.*)</primary(.*)");
+    httpAttrMap.put("secondary", "(.*)secondary>(.*)</secondary(.*)");
+    httpAttrMap.put("count", "(.*)countattribute>(.*)</countattribute(.*)");
+  }
+
+  @AfterAll
+  void tearDown() {
+  }
+
+  @ParameterizedTest
+  @MethodSource("getOrigServerAndSessionInfo")
+  @DisplayName("Stop the primary server, verify that a new primary server is picked and HTTP session state is migrated")
```

Comment:
Provide the javadoc descrbing the usecase detail

Reply:
Added